geps/gep-713/index.md

```diff
@@ -1,1157 +1,1082 @@
+* Provide a consistent specification that will ensure familiarity between both included and implementation-specific metaresources so they can both be interpreted the same way.
+* Provide a reference pattern to other implementations of metaresource and policy APIs outside of Gateway API, that are based on similar concepts (i.e., augmenting the behavior of other Kubernetes objects, attachment points, nested contexts and inheritance, Defaults & Overrides, etc.)
 
 ## Out of scope
 
-* Define all potential policies that may be attached to resources
-* Design the full structure and configuration of policies
+* Define all potential metaresource and policy kinds that may be attached to resources.
+
+## Background
+
+When designing Gateway API, a recurring challenge became apparent. There was often a need to change the behavior of objects without modifying their specs. Sometimes, this is because changing the spec of the object to hold the new piece of information is not possible (e.g., `ReferenceGrant`, from [GEP-709](https://gateway-api.sigs.k8s.io/geps/gep-709/), when affecting Secrets and Services), and sometimes it’s because the behavior change is intended to flow across multiple objects (see [Semantics](#semantics-why) of metaresources and [Inherited](#inherited) class of metaresources).
+
+To put this another way, sometimes we need ways to be able to affect how an object is interpreted in the API, without representing the description of those effects inside the spec of the object. This document describes the ways to design objects to meet use cases like these.
+
+This document introduces the concept of a “metaresource”, a term used to describe the class of objects that _only_ augment the behavior of another Kubernetes object, regardless of what they are targeting.
+
+“Meta” here is used in its Greek sense of “more comprehensive” or “transcending”, and “resource” rather than “object” because “metaresource” is more pronounceable than “meta object”.
+
+After a few iterations of Gateway API experimenting with this pattern－both, with its own common kinds of metaresources such as the `BackendTLSPolicy`, as well as via multiple implementation-specific kinds of metaresources (see examples of [current use of metaresources](#current-use-of-metaresources))－ and rounds of discussion such as the one at [kubernetes-sigs/gateway-api/discussions#2927](https://github.com/kubernetes-sigs/gateway-api/discussions/2927), the pattern has been enhanced to its current form.
+
+## Definition
+
+### Metaresources
+
+*Metaresources* are objects that augment the behavior of other objects (*targets*) in a clean and standard way, by declaring additional specification for these targets “from the outside”－i.e., without directly modifying the objects whose behavior the metaresource intends to affect, nor requiring any reference from the objects to the metaresources that affect them.
+
+Metaresources follow a well-known structure that declares clearly a *target* and an *intent*.
```

Comment:
Could also be considered a Decorator pattern.

Reply:
Agreed. I'll incorporate it in the text. It may indeed help people with an OOP background understand the concept. Thanks!

Reply:
Hmmm. Actually, now having second thoughts... By reading "decorator pattern", could some people inadvertently think a metaresource "wraps the target", as in it behaves like the object it targets except for the decoration it adds on top? That would be the wrong message. WDYT?

Reply:
How about adding a reference to the specific pattern you're talking about here, or if one can't be found then state that it is a new and necessary pattern needed to enhance expressiveness in Gateway API?

Reply:
I guess the [Composite pattern](https://en.wikipedia.org/wiki/Composite_pattern) is probably a closer fit?